src/Compilers/VisualBasic/Portable/Language/LanguageFeature.vb

```diff
@@ -0,0 +1,288 @@
+        <Extension>
+        Friend Function IsAvailable(feature As Feature, options As VisualBasicParseOptions) As Boolean
+            Return feature.IsInLanguageVersion(options) OrElse feature.CheckFeatures(options)
+        End Function
+
+        ''' <summary>
+        ''' Check to see if a language <paramref name="feature"/> is available with these <paramref name="compilation"/>.
+        ''' </summary>
+        ''' <param name="feature">Language feature to check is available.</param>
+        ''' <param name="compilation">The <see cref="VisualBasicCompilation"/> being used.</param>
+        <Extension>
+        Friend Function IsAvailable(feature As Feature, compilation As VisualBasicCompilation) As Boolean
+            Return feature.IsInLanguageVersion(compilation)
+        End Function
+
+        ''' <summary>
+        ''' Check to see if a language <paramref name="feature"/> is available with the <paramref name="options"/> being used.
+        ''' If unavailable the function return the node with an unavailable diagnostic attached to <paramref name="node"/>.
+        ''' </summary>
+        ''' <returns>If <see cref="feature"/> is not available, returns node with unavailable diagnostic attached to it.</returns>
+        ''' <param name="node">The node to attach the potential diagnostic (Feature Unavailable).</param>
+        ''' <param name="feature">Language feature to check is available.</param>
+        ''' <param name="options">The parse options being used.</param>
+        <Extension>
+        Friend Function CheckFeatureAvailability(Of TNode As Syntax.InternalSyntax.VisualBasicSyntaxNode)(node As TNode, feature As Feature, options As VisualBasicParseOptions) As TNode
+            Return If(feature.IsAvailable(options), node, node.ReportFeatureUnavailable(feature, options))
+        End Function
+
+        ''' <summary>
+        ''' Checks to see if a language <paramref name="feature"/> is unavailable to use with the <paramref name="options"/> being used.
+        ''' If unavailable the function returns False and adds an unavailable diagnostic to the <paramref name="diagnostics"/> at <paramref name="location"/>.
+        ''' </summary>
+        ''' <returns>
+        ''' False: Feature is available.
+        '''  True: Feature is unavailable, and a unavailable diagnostic is add to <paramref name="diagnostics"/> at the <paramref name="location"/>.
+        ''' </returns>
+        ''' <param name="feature">Language feature to check is available.</param>
+        ''' <param name="options">The parse options being used.</param>
+        ''' <param name="diagnostics">The diagnostics to which add this diagnostic.</param>
+        ''' <param name="location">The location to report the diagnostic.</param>
+        <Extension>
+        Friend Function IsUnavailable(feature As Feature, options As VisualBasicParseOptions, diagnostics As DiagnosticBag, location As Location) As Boolean
```

Comment:
The feedback on naming this method was that an `Is...` method does not clearly convey side-effects (to diagnostics). The original name, `CheckFeatureAvailability`, makes more sense here. Thanks